### DISCUSSION TITLE
Blender setup world script

### DISCUSSION THREAD

**Discussion by michal212345** [BramStoutProductions/MiEx#32] [Show and tell]
Hello wanted to share this small blender script to set up worlds imported with MiEx and USDPreviewSurfaces, this is a small patch until Blender decides to add MaterialX

I recreated the nodes that add block tints.

Small render to prove it works:

![untitled-min](https://github.com/BramStoutProductions/MiEx/assets/20019071/e9d7ad1d-2b32-49ad-9025-e64ed7ed5dea)

Link to the script:
[https://pastebin.com/bwnhDpxd](https://pastebin.com/bwnhDpxd)

**Comment by BlueEvilGFX** [comment]
Hi michal, 
i am currently writing a small addon for exactly that! Is it okay for you that i re-use  `def block_side_adjustment(...)`? I would name you as one of the authors.
![image](https://github.com/BramStoutProductions/MiEx/assets/107540992/d0498436-2db7-445e-ab36-ad7c077e3e31)

There are also differences between @michal212345 and my script. Might even try to commit it to the main branch if a blender addon is desired.

**Reply by michal212345** [reply]
Sure!

**Comment by StandingPadAnimations** [comment]
MCprep developer here, MiEx has some basic support in MCprep, albeit unintentional (prep materials works, but I doubt stuff like mesh swap will). If you're willing @BlueEvilGFX and @michal212345, perhaps we could work together on supporting MiEx specific stuff :D 

EDIT: We have a issue tracking MiEx support on the MCprep repo: https://github.com/Moo-Ack-Productions/MCprep/issues/563

**Reply by BlueEvilGFX** [reply]
This might be a bit above my level ¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯

**Reply by michal212345** [reply]
Tbh, I just made this as a temporary fix until maybe Blender supports MaterialX, but I will definitely look into contributing some form of semi-permanent fix @StandingPadAnimations 😄
